### PR TITLE
Add package testing

### PR DESCRIPTION
**Pull request by johnbieren**
This adds the standard-test-roles testing into the pipeline as its own stage. The stage is a fully containerized, OpenShift hosted container call.

### CODE REVIEW

**Comment by johnbieren** [comment]
[test]

**Comment by johnbieren** [comment]
[test]

File: src/org/centos/pipeline/PipelineUtils.groovy [hunk 1]
```diff
@@ -471,6 +474,9 @@ def rsyncData(String stage){
                 "export image2boot=\"${env.image2boot}\"\n" +
                 "export commit=\"${env.commit}\"\n" +
                 "export ANSIBLE_HOST_KEY_CHECKING=\"False\"\n"
+    } else if (stage == 'ci-pipeline-functional-tests') {
+        text = text +
+                "export package=\"${env.fed_repo}\"\n" +
```
Comment (scoheb) [inline src/org/centos/pipeline/PipelineUtils.groovy]: extra '+' at end...needs to be removed

**Comment by johnbieren** [comment]
[test]

File: JenkinsfileStageTrigger [hunk 2]
```diff
@@ -107,6 +108,17 @@ pipeline {
                 }
             }
         }
+        stage("package-test image build") {
+            when {
+                // Only build if we have related files in changeset
+                changeset "config/Dockerfiles/package-test/**"
```
Comment (scoheb) [inline JenkinsfileStageTrigger]: '-' should be '_' as it is in file-system

**Comment by johnbieren** [comment]
[test]

**Comment by johnbieren** [comment]
[test] curl hang up

**Comment by johnbieren** [comment]
[test] weird ostree boot sanity failure

**Comment by johnbieren** [comment]
[test] but do not merge, forcing it to run some tests, have to remove a comment or two if it passes before merging

**Comment by johnbieren** [comment]
[test]

**Comment by johnbieren** [comment]
[test]

**Comment by johnbieren** [comment]
[test] ostree compose error

**Comment by johnbieren** [comment]
[merge] Passed stage pipeline, see https://jenkins-continuous-infra.apps.ci.centos.org/view/CI-Stage-Pipeline/job/ci-stage-pipeline-f26/339/

**Comment by johnbieren** [comment]
@stefwalter FYI. This + https://github.com/CentOS-PaaS-SIG/ci-pipeline/pull/395 has the standard-test-roles testing in the pipeline as its own stage, on a fresh qcow2 image each time.  The test logs are archived by Jenkins as well as stored in the logs dir on artifacts.ci.centos.org.  The one thing to note is that many of the packages with tests only have them in master, not f26, which we are currently building against.

**Comment by stefwalter** [comment]
Thanks @johnbieren. Nice!

I'll poke @dperpeet about master vs f26.

**Pull request merged**